pkg/work/spoke/apply/update_apply.go

```diff
@@ -172,3 +201,46 @@ func isSameUnstructured(obj1, obj2 *unstructured.Unstructured) bool {
 
 	return equality.Semantic.DeepEqual(obj1Copy.Object, obj2Copy.Object)
 }
+
+// detect changes in a resource by caching a hash of the string representation of the resource
+// note: some changes in a resource e.g. nil vs empty, will not be detected this way
+func hashOfResourceStruct(o interface{}) string {
+	oString := fmt.Sprintf("%v", o)
+	h := md5.New()
+	io.WriteString(h, oString)
+	rval := fmt.Sprintf("%x", h.Sum(nil))
+	return rval
+}
+
+type objectHashCache struct{}
+
+func (c *objectHashCache) UpdateCachedResourceMetadata(_ runtime.Object, _ runtime.Object) {
+	return
+}
+
+func (c *objectHashCache) SafeToSkipApply(required runtime.Object, existing runtime.Object) bool {
+	requiredAccessor, err := meta.Accessor(required)
+	if err != nil {
+		return true
+	}
+	// existing could be nil in some cases.
+	if existing == nil {
+		return false
+	}
+	existingAccessor, err := meta.Accessor(existing)
+	if err != nil {
+		return true
```

Comment:
if building accessor for the existing object failed, should we apply the object?